### PR TITLE
baseplate: Add StopDelay to Config

### PR DESCRIPTION
**Pull request by fishy**
This helps mitigate against a race condition from kubernetes between
draining the traffic from the pod and sending the pod SIGTERM.

### CODE REVIEW

File: baseplate.go [hunk 1]
```diff
@@ -17,6 +17,11 @@ import (
 	"github.com/reddit/baseplate.go/tracing"
 )
 
+const (
+	// DefaultStopDelay is the default StopDelay to be used in Serve.
+	DefaultStopDelay = time.Second
```
Comment (nanassito) [inline baseplate.go:22]: I know it doesn't change anything but it is easier to read IMO
```suggestion
	DefaultStopDelay = 1 * time.Second
```

**Review by nanassito — APPROVED** [review]

File: baseplate.go (hunk 1 above)
Comment (kylelemons) [reply baseplate.go:22]: +1

File: batchcloser/closers.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ type BatchCloser struct {
 func (bc *BatchCloser) Close() error {
 	var errs errorsbp.Batch
 	for _, closer := range bc.closers {
-		errs.AddPrefix(fmt.Sprintf("%#v", closer), closer.Close())
+		errs.AddPrefix(fmt.Sprintf("%T", closer), closer.Close())
```
Comment (kylelemons) [inline batchcloser/closers.go:52]: 🔕 (this is a bit out of scope, and it is a somewhat major reduction in the verbosity of the prefix, might be worth not doing this in this PR at least)

**Review by kylelemons — APPROVED** [review]

File: batchcloser/closers.go (hunk 2 above)
Comment (fishy) [reply batchcloser/closers.go:52]: Not doing this will cause a race condition in the `timestampCloser` used in tests, which seems to be a new issue in go 1.19 (we don't see this race condition from before)

I'll add a note in the PR description/commit message.

Comment (fishy) [reply batchcloser/closers.go]: ok I reverted this and implemented `GoString` on `timestampCloser` instead.

**Review by kylelemons — APPROVED** [review]
Aha! I'd just mention the 1.19 fix when you squash+merge

**Comment by fishy** [comment]
> Aha! I'd just mention the 1.19 fix when you squash+merge

the fix is purely in the unit test code so I don't think it's worth mentioning.

**Review by pacejackson — APPROVED** [review]
